From: Mangal Kushwah <mangalk2324@gmail.com>
Date: Fri, 9 Aug 2024 19:30:05 +0530
Subject: codec: videotoolbox: remove sw decoder fallback for mp4v

commit 7a2bf498d545d500d30636970fa930eb54de5569 insert packetizer for
xvid mpeg4 video, so now hw decoder can decode xvid encoded video
without any playback issues.

(cherry picked from commit a9343f68bb19a1c20e45e1029848326020b0c69e)
---
 modules/codec/videotoolbox.m | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/modules/codec/videotoolbox.m b/modules/codec/videotoolbox.m
index e07d3c6..7d408bf 100644
--- a/modules/codec/videotoolbox.m
+++ b/modules/codec/videotoolbox.m
@@ -1024,11 +1024,6 @@ static CMVideoCodecType CodecPrecheck(decoder_t *p_dec)
 
         case VLC_CODEC_MP4V:
         {
-            if (p_dec->fmt_in.i_original_fourcc == VLC_FOURCC( 'X','V','I','D' )) {
-                msg_Warn(p_dec, "XVID decoding not implemented, fallback on software");
-                return -1;
-            }
-
             msg_Dbg(p_dec, "Will decode MP4V with original FourCC '%4.4s'", (char *)&p_dec->fmt_in.i_original_fourcc);
             return kCMVideoCodecType_MPEG4Video;
         }
